### DISCUSSION TITLE
Flex 와 Grid 의 차이점

### DISCUSSION THREAD

**Discussion by WooGie911** [wanted-9th-3team/frontend-study#46] [프론트 기술 질문 공간] [answered]
Flex 와 Grid 의 차이점이 무엇인가요?
Flex 와 Grid 에 대해 간단하게 설명해주세요.

**Comment by michoball** [comment]
가장 큰 차이점은 flex 는 1차원으로 레이아웃을 다루는데 적합하고 grid는 2차원으로 다루는데 적합하다는 점입니다. 

flex는 기본적으로 row 방향으로 레이아웃을 만들고 column 방향으로는 따로 direction을 지정해줘야합니다. 
반면 grid는 grid도 정의된 구역 내에서 row 와 column을 둘다 다룰 수 있습니다.

**Comment by WooGie911** [comment, marked as answer]
둘의 차이라면 Flex 방식은 한 방향으로 레이아웃하는 1차원 방식이라고 할 수 있고, 
Grid 방식은 두 방향 레이아웃을 배치할 수 있는 2차원 방식이라는 차이점입니다.

Flex는 1차원으로 수평, 수직 영역 중 하나의 방향으로만 레이아웃을 나눌 수 있고, 
Grid는 2차원으로 수평 수직을 동시에 영역을 나눌 수 있습니다.

요약하면 Flex는 콘텐츠 위주의 정렬, Grid는 레이아웃 위주의 정렬이라고 할 수 있습니다.

따라서 이미 결정된 레이아웃을 만들고 싶으면 Grid 속성을 사용하고, 자유로운 레이아웃을 만들고 싶으면 Flex 속성을 사용하면 됩니다.

**Comment by BaekYuri** [comment]
Flex는 1차원 방식으로 레이아웃을 나누고, Grid는 2차원 방식으로 레이아웃을 나눕니다.
Flex는 Flex 컨테이너 안에 있는 태그들의 방향을 설정할 수 있고, flex-direction 속성으로 수평, 수직 설정을 할 수 있습니다.
Grid는 수평 수직을 동시에 영역을 나눌 수 있습니다. grid-template-columns, grid-template-rows 속성을 이용해 행과 열을 설정할 수 있습니다.
